### ISSUE TITLE
[loc] The custom configure preset is not translated in the VS Code Command Palette drop-down options

### ISSUE THREAD

**Issue by v-frankwang** [microsoft/vscode-cmake-tools#3803]
### Brief Issue Summary

Repro steps:

1.   Create a new folder named 'Test' and open it with vscode.
2.   Create a file named CMakeLists.txt with the following content:    
 ```
         cmake_minimum_required(VERSION 3.8) 
         project(Test)
         add_executable(Test "main.cpp")

 ```
3.   Create a file named test.cpp with the following content:    
```
         #include <iostream> 
         auto main() -> int { std::cout << "hello world" << std::endl; }
```
4.   Create a file named CMakePresets.json with the following content:
 ```
  {
             "version": 3,
             "configurePresets": [
                   {
                        "name": "windows-base",
                        "hidden": true,
                        "generator": "Ninja",
                        "binaryDir": "${sourceDir}/out/build/${presetName}",
                        "installDir": "${sourceDir}/out/install/${presetName}",
                        "cacheVariables": {
                        "CMAKE_C_COMPILER": "cl.exe",
                        "CMAKE_CXX_COMPILER": "cl.exe"
                        },
                       "condition": {
                       "type": "equals",
                       "lhs": "${hostSystemName}",
                       "rhs": "Windows"
                     }
                  },
                 {
                      "name": "x64-debug",
                      "displayName": "x64 Debug",
                      "inherits": "windows-base",
                      "architecture": {
                      "value": "x64",
                       "strategy": "external"
                  },
                       "cacheVariables": {
                        "CMAKE_BUILD_TYPE": "Debug"
                 }
              },
                   {
                       "name": "x64-release",
                       "displayName": "x64 Release",
                       "inherits": "x64-debug",
                       "cacheVariables": {
                       "CMAKE_BUILD_TYPE": "Release"
                }
             },
            {
                  "name": "x86-debug",
                  "displayName": "x86 Debug",
                  "inherits": "windows-base",
                  "architecture": {
                  "value": "x86",
                  "strategy": "external"
             },
                  "cacheVariables": {
                  "CMAKE_BUILD_TYPE": "Debug"
               }
               },
               {
                   "name": "x86-release",
                   "displayName": "x86 Release",
                   "inherits": "x86-debug",
                   "cacheVariables": {
                   "CMAKE_BUILD_TYPE": "Release"
                 }
            }
        ]
  }
```

5 .  Run command: “**CMake:configure**”
6.   Run command: “**CMake: Add configure preset**”
7.   Select “**custom**”
8.   Enter the name of the preset:”**Custom configure preset**”
9.   Run command: “**CMake:Select configure preset**” 
10. Watch drop-down list

Actual results：
![image](https://github.com/microsoft/vscode-cmake-tools/assets/160998958/e2acbcea-4432-4e62-bf3d-a458c38c07b1)

**Comment by v-frankwang** [comment]
@gcampbell-msft I verified the issue on the latest version：v1.18.42 of CMake tools and it still reproduced, so I reopen this issue.

Actual results:
![image](https://github.com/microsoft/vscode-cmake-tools/assets/160998958/fa6d0a22-2aa8-4dd2-ada6-50a5dd254136)

**Comment by v-frankwang** [comment]
@gcampbell-msft There is a similar issue with this issue, you can get the details below.

**Repro steps:**
1. Create a new folder named 'Test' and open it with vscode.
2. Create a file named CMakeLists.txt with the following content:
```
cmake_minimum_required(VERSION 3.8) 
project(Test)
add_executable(Test "main.cpp")
```
3. Create a file named test.cpp with the following content:
```
#include <iostream> 
auto main() -> int { std::cout << "hello world" << std::endl; }
```
4. Create a file named CMakePresets.json with the following content:
```
{ 
    "version": 2, 
    "configurePresets": [ 
        { 
            "name": "windows-default", 
            "displayName": "Windows x64 Debug", 
            "description": "Sets Ninja generator, compilers, x64 architecture, build and install directory, debug build type", 
            "generator": "Ninja", 
            "binaryDir": "${sourceDir}/out/build/${presetName}", 
            "architecture": { 
                "value": "x64", 
                "strategy": "external" 
            }, 
            "cacheVariables": { 
                "CMAKE_BUILD_TYPE": "Debug", 
                "CMAKE_INSTALL_PREFIX": "${sourceDir}/out/install/${presetName}" 
            }, 
            "vendor": { 
                "microsoft.com/VisualStudioSettings/CMake/1.0": { 
                    "hostOS": [ 
                        "Windows" 
                    ] 
                } 
            } 
        } 
    ] 
} 
```
5. Run command: “**CMake:Select configure preset**”

**Actual result:** 
![image](https://github.com/microsoft/vscode-cmake-tools/assets/160998958/4bae1ae8-b095-4095-b1a6-5e2c6f3c34bf)

**Comment by Ashley-Li** [comment]
Verified on CMake Tools v1.18.43(pre-release), this issue still repro:
![image](https://github.com/microsoft/vscode-cmake-tools/assets/56664234/0b12560d-2423-4b0e-8b10-d8ffaae02e62)
![image](https://github.com/microsoft/vscode-cmake-tools/assets/56664234/62a6c54e-8992-4ce7-9cf8-27ab66c2782c)

**Comment by v-frankwang** [comment]
@gcampbell-msft We verified this issue on CMake Tools: v1.20.3 (pre-release). When we set the language of vscode to Chinese, the above similar problem: https://github.com/microsoft/vscode-cmake-tools/issues/3803#issuecomment-2201696036 will still reappear. We also found a new issue that when the language of vscode is reset back to English, in the drop-down list of CMake: Select configure preset, the Preset added in Chinese state is still displayed in Chinese. Do we need to reopen this issue or create a new issue to track it?

[CMake Tools Diagnostics.txt](https://github.com/user-attachments/files/16846166/CMake.Tools.Diagnostics.txt)

Actual result:
![3803](https://github.com/user-attachments/assets/da88abf7-9f4b-4f06-84b2-df42bbb8b788)
